### ISSUE TITLE
invalid function org-with-silent-modifications 报这个错，请问是什么原因？

### ISSUE THREAD

**Issue by dreamer-yzy** [kelvinh/org-page#136]

**Comment by kelvinh** [comment]
Please provide more detailed info, like Emacs/org-mode version, and also the call stack when the error happens.

**Comment by trentfisher** [comment]
Well, I got this exact same error, here are my details:

Emacs: This is GNU Emacs 24.3.1 (i686-redhat-linux-gnu, GTK+ Version 3.10.9) of 2014-09-30 on buildhw-01.phx2.fedoraproject.org

Org-mode version 8.2.10 (8.2.10-37-gc000e8-elpa @ /home/trent/.emacs.d/elpa/org-20150427/)

I installed from Melpa per the instructions (which got me a newer org-mode as well)

I used op/new-repository, and then op-newpost, and then ran (op/do-publication t nil nil t)

Here's the stack trace:

Debugger entered--Lisp error: (invalid-function org-with-silent-modifications)
  org-with-silent-modifications(nil)
  org-refresh-category-properties()
  org-get-category()
  org-entry-properties(nil standard)
  org-element-headline-parser(90 nil)
  org-element--current-element(90 nil first-section nil)
  org-element--parse-elements(1 90 first-section nil nil nil (org-data nil (section (:begin 1 :end 17 :contents-begin 1 :contents-end 16 :post-blank 1 :parent #0) (keyword (:key "TITLE" :value "About" :begin 1 :end 16 :post-blank 0 :post-affiliated 1 :parent #1)))))
  org-element-parse-buffer(nil nil)
  (org-export-filter-apply-functions (plist-get info :filter-parse-tree) (org-element-parse-buffer nil visible-only) info)
  (setq tree (org-export-filter-apply-functions (plist-get info :filter-parse-tree) (org-element-parse-buffer nil visible-only) info))
  (progn (run-hook-with-args (quote org-export-before-processing-hook) (progn (or (and (vectorp backend) (>= (length backend) 8) (memq (aref backend 0) cl-struct-org-export-backend-tags)) (error "%s accessing a non-%s" (quote org-export-backend-name) (quote org-export-backend))) (aref backend 1))) (org-export-expand-include-keyword) (org-macro-initialize-templates) (org-macro-replace-all org-macro-templates) (org-export-execute-babel-code) (org-update-radio-target-regexp) (goto-char (point-min)) (save-excursion (run-hook-with-args (quote org-export-before-parsing-hook) (progn (or (and (vectorp backend) (>= (length backend) 8) (memq (aref backend 0) cl-struct-org-export-backend-tags)) (error "%s accessing a non-%s" (quote org-export-backend-name) (quote org-export-backend))) (aref backend 1)))) (setq info (org-export-install-filters (org-combine-plists info (org-export-get-environment backend subtreep ext-plist)))) (if (plist-get info :title) nil (plist-put info :title (let ((file (plist-get info :input-file))) (if file (file-name-sans-extension (file-name-nondirectory file)) (plist-get info :input-buffer))))) (org-macro-replace-all (list (cons "author" (org-element-interpret-data (plist-get info :author))) (cons "date" (org-element-interpret-data (plist-get info :date))) (cons "email" (or (plist-get info :email) "")) (cons "title" (org-element-interpret-data (plist-get info :title))))) (let ((backend-name (progn (or (and (vectorp backend) (>= ... 8) (memq ... cl-struct-org-export-backend-tags)) (error "%s accessing a non-%s" (quote org-export-backend-name) (quote org-export-backend))) (aref backend 1)))) (progn (let ((--dolist-tail-- (plist-get info :filter-options)) filter) (while --dolist-tail-- (setq filter (car --dolist-tail--)) (let ((result ...)) (if result (progn ...))) (setq --dolist-tail-- (cdr --dolist-tail--)))))) (setq tree (org-export-filter-apply-functions (plist-get info :filter-parse-tree) (org-element-parse-buffer nil visible-only) info)) (setq info (org-combine-plists info (org-export-collect-tree-properties tree info))) (let* ((body (org-element-normalize-string (or (org-export-data tree info) ""))) (inner-template (cdr (assq (quote inner-template) (plist-get info :translate-alist)))) (full-body (if (not (functionp inner-template)) body (funcall inner-template body info))) (template (cdr (assq (quote template) (plist-get info :translate-alist))))) (org-no-properties (org-export-filter-apply-functions (plist-get info :filter-final-output) (if (or (not (functionp template)) body-only) full-body (funcall template full-body info)) info))))
  (save-current-buffer (set-buffer --buf-copy) (goto-char (point-min)) (progn (run-hook-with-args (quote org-export-before-processing-hook) (progn (or (and (vectorp backend) (>= (length backend) 8) (memq (aref backend 0) cl-struct-org-export-backend-tags)) (error "%s accessing a non-%s" (quote org-export-backend-name) (quote org-export-backend))) (aref backend 1))) (org-export-expand-include-keyword) (org-macro-initialize-templates) (org-macro-replace-all org-macro-templates) (org-export-execute-babel-code) (org-update-radio-target-regexp) (goto-char (point-min)) (save-excursion (run-hook-with-args (quote org-export-before-parsing-hook) (progn (or (and (vectorp backend) (>= ... 8) (memq ... cl-struct-org-export-backend-tags)) (error "%s accessing a non-%s" (quote org-export-backend-name) (quote org-export-backend))) (aref backend 1)))) (setq info (org-export-install-filters (org-combine-plists info (org-export-get-environment backend subtreep ext-plist)))) (if (plist-get info :title) nil (plist-put info :title (let ((file (plist-get info :input-file))) (if file (file-name-sans-extension (file-name-nondirectory file)) (plist-get info :input-buffer))))) (org-macro-replace-all (list (cons "author" (org-element-interpret-data (plist-get info :author))) (cons "date" (org-element-interpret-data (plist-get info :date))) (cons "email" (or (plist-get info :email) "")) (cons "title" (org-element-interpret-data (plist-get info :title))))) (let ((backend-name (progn (or (and ... ... ...) (error "%s accessing a non-%s" ... ...)) (aref backend 1)))) (progn (let ((--dolist-tail-- (plist-get info :filter-options)) filter) (while --dolist-tail-- (setq filter (car --dolist-tail--)) (let (...) (if result ...)) (setq --dolist-tail-- (cdr --dolist-tail--)))))) (setq tree (org-export-filter-apply-functions (plist-get info :filter-parse-tree) (org-element-parse-buffer nil visible-only) info)) (setq info (org-combine-plists info (org-export-collect-tree-properties tree info))) (let* ((body (org-element-normalize-string (or (org-export-data tree info) ""))) (inner-template (cdr (assq (quote inner-template) (plist-get info :translate-alist)))) (full-body (if (not (functionp inner-template)) body (funcall inner-template body info))) (template (cdr (assq (quote template) (plist-get info :translate-alist))))) (org-no-properties (org-export-filter-apply-functions (plist-get info :filter-final-output) (if (or (not ...) body-only) full-body (funcall template full-body info)) info)))))
  (unwind-protect (save-current-buffer (set-buffer --buf-copy) (goto-char (point-min)) (progn (run-hook-with-args (quote org-export-before-processing-hook) (progn (or (and (vectorp backend) (>= ... 8) (memq ... cl-struct-org-export-backend-tags)) (error "%s accessing a non-%s" (quote org-export-backend-name) (quote org-export-backend))) (aref backend 1))) (org-export-expand-include-keyword) (org-macro-initialize-templates) (org-macro-replace-all org-macro-templates) (org-export-execute-babel-code) (org-update-radio-target-regexp) (goto-char (point-min)) (save-excursion (run-hook-with-args (quote org-export-before-parsing-hook) (progn (or (and ... ... ...) (error "%s accessing a non-%s" ... ...)) (aref backend 1)))) (setq info (org-export-install-filters (org-combine-plists info (org-export-get-environment backend subtreep ext-plist)))) (if (plist-get info :title) nil (plist-put info :title (let ((file ...)) (if file (file-name-sans-extension ...) (plist-get info :input-buffer))))) (org-macro-replace-all (list (cons "author" (org-element-interpret-data (plist-get info :author))) (cons "date" (org-element-interpret-data (plist-get info :date))) (cons "email" (or (plist-get info :email) "")) (cons "title" (org-element-interpret-data (plist-get info :title))))) (let ((backend-name (progn (or ... ...) (aref backend 1)))) (progn (let ((--dolist-tail-- ...) filter) (while --dolist-tail-- (setq filter ...) (let ... ...) (setq --dolist-tail-- ...))))) (setq tree (org-export-filter-apply-functions (plist-get info :filter-parse-tree) (org-element-parse-buffer nil visible-only) info)) (setq info (org-combine-plists info (org-export-collect-tree-properties tree info))) (let* ((body (org-element-normalize-string (or ... ""))) (inner-template (cdr (assq ... ...))) (full-body (if (not ...) body (funcall inner-template body info))) (template (cdr (assq ... ...)))) (org-no-properties (org-export-filter-apply-functions (plist-get info :filter-final-output) (if (or ... body-only) full-body (funcall template full-body info)) info))))) (and (buffer-live-p --buf-copy) (progn (save-current-buffer (set-buffer --buf-copy) (restore-buffer-modified-p nil)) (kill-buffer --buf-copy))))
  (let ((--buf-copy (org-export-copy-buffer))) (unwind-protect (save-current-buffer (set-buffer --buf-copy) (goto-char (point-min)) (progn (run-hook-with-args (quote org-export-before-processing-hook) (progn (or (and ... ... ...) (error "%s accessing a non-%s" ... ...)) (aref backend 1))) (org-export-expand-include-keyword) (org-macro-initialize-templates) (org-macro-replace-all org-macro-templates) (org-export-execute-babel-code) (org-update-radio-target-regexp) (goto-char (point-min)) (save-excursion (run-hook-with-args (quote org-export-before-parsing-hook) (progn (or ... ...) (aref backend 1)))) (setq info (org-export-install-filters (org-combine-plists info (org-export-get-environment backend subtreep ext-plist)))) (if (plist-get info :title) nil (plist-put info :title (let (...) (if file ... ...)))) (org-macro-replace-all (list (cons "author" (org-element-interpret-data ...)) (cons "date" (org-element-interpret-data ...)) (cons "email" (or ... "")) (cons "title" (org-element-interpret-data ...)))) (let ((backend-name (progn ... ...))) (progn (let (... filter) (while --dolist-tail-- ... ... ...)))) (setq tree (org-export-filter-apply-functions (plist-get info :filter-parse-tree) (org-element-parse-buffer nil visible-only) info)) (setq info (org-combine-plists info (org-export-collect-tree-properties tree info))) (let* ((body (org-element-normalize-string ...)) (inner-template (cdr ...)) (full-body (if ... body ...)) (template (cdr ...))) (org-no-properties (org-export-filter-apply-functions (plist-get info :filter-final-output) (if ... full-body ...) info))))) (and (buffer-live-p --buf-copy) (progn (save-current-buffer (set-buffer --buf-copy) (restore-buffer-modified-p nil)) (kill-buffer --buf-copy)))))
  (let* ((org-export-current-backend (progn (or (and (vectorp backend) (>= (length backend) 8) (memq (aref backend 0) cl-struct-org-export-backend-tags)) (error "%s accessing a non-%s" (quote org-export-backend-name) (quote org-export-backend))) (aref backend 1))) (info (org-combine-plists (list :export-options (delq nil (list (and subtreep ...) (and visible-only ...) (and body-only ...)))) (org-export--get-buffer-attributes))) tree) (let ((--buf-copy (org-export-copy-buffer))) (unwind-protect (save-current-buffer (set-buffer --buf-copy) (goto-char (point-min)) (progn (run-hook-with-args (quote org-export-before-processing-hook) (progn (or ... ...) (aref backend 1))) (org-export-expand-include-keyword) (org-macro-initialize-templates) (org-macro-replace-all org-macro-templates) (org-export-execute-babel-code) (org-update-radio-target-regexp) (goto-char (point-min)) (save-excursion (run-hook-with-args (quote org-export-before-parsing-hook) (progn ... ...))) (setq info (org-export-install-filters (org-combine-plists info ...))) (if (plist-get info :title) nil (plist-put info :title (let ... ...))) (org-macro-replace-all (list (cons "author" ...) (cons "date" ...) (cons "email" ...) (cons "title" ...))) (let ((backend-name ...)) (progn (let ... ...))) (setq tree (org-export-filter-apply-functions (plist-get info :filter-parse-tree) (org-element-parse-buffer nil visible-only) info)) (setq info (org-combine-plists info (org-export-collect-tree-properties tree info))) (let* ((body ...) (inner-template ...) (full-body ...) (template ...)) (org-no-properties (org-export-filter-apply-functions ... ... info))))) (and (buffer-live-p --buf-copy) (progn (save-current-buffer (set-buffer --buf-copy) (restore-buffer-modified-p nil)) (kill-buffer --buf-copy))))))
  (save-restriction (cond ((org-region-active-p) (narrow-to-region (region-beginning) (region-end))) (subtreep (org-narrow-to-subtree) (goto-char (point-min)) (forward-line) (narrow-to-region (point) (point-max)))) (let* ((org-export-current-backend (progn (or (and (vectorp backend) (>= ... 8) (memq ... cl-struct-org-export-backend-tags)) (error "%s accessing a non-%s" (quote org-export-backend-name) (quote org-export-backend))) (aref backend 1))) (info (org-combine-plists (list :export-options (delq nil (list ... ... ...))) (org-export--get-buffer-attributes))) tree) (let ((--buf-copy (org-export-copy-buffer))) (unwind-protect (save-current-buffer (set-buffer --buf-copy) (goto-char (point-min)) (progn (run-hook-with-args (quote org-export-before-processing-hook) (progn ... ...)) (org-export-expand-include-keyword) (org-macro-initialize-templates) (org-macro-replace-all org-macro-templates) (org-export-execute-babel-code) (org-update-radio-target-regexp) (goto-char (point-min)) (save-excursion (run-hook-with-args ... ...)) (setq info (org-export-install-filters ...)) (if (plist-get info :title) nil (plist-put info :title ...)) (org-macro-replace-all (list ... ... ... ...)) (let (...) (progn ...)) (setq tree (org-export-filter-apply-functions ... ... info)) (setq info (org-combine-plists info ...)) (let* (... ... ... ...) (org-no-properties ...)))) (and (buffer-live-p --buf-copy) (progn (save-current-buffer (set-buffer --buf-copy) (restore-buffer-modified-p nil)) (kill-buffer --buf-copy)))))))
  (save-excursion (save-restriction (cond ((org-region-active-p) (narrow-to-region (region-beginning) (region-end))) (subtreep (org-narrow-to-subtree) (goto-char (point-min)) (forward-line) (narrow-to-region (point) (point-max)))) (let* ((org-export-current-backend (progn (or (and ... ... ...) (error "%s accessing a non-%s" ... ...)) (aref backend 1))) (info (org-combine-plists (list :export-options (delq nil ...)) (org-export--get-buffer-attributes))) tree) (let ((--buf-copy (org-export-copy-buffer))) (unwind-protect (save-current-buffer (set-buffer --buf-copy) (goto-char (point-min)) (progn (run-hook-with-args ... ...) (org-export-expand-include-keyword) (org-macro-initialize-templates) (org-macro-replace-all org-macro-templates) (org-export-execute-babel-code) (org-update-radio-target-regexp) (goto-char ...) (save-excursion ...) (setq info ...) (if ... nil ...) (org-macro-replace-all ...) (let ... ...) (setq tree ...) (setq info ...) (let* ... ...))) (and (buffer-live-p --buf-copy) (progn (save-current-buffer ... ...) (kill-buffer --buf-copy))))))))
  org-export-as(html nil nil t nil)
  (let ((--cl-org-html-fontify-code-- (function (lambda (code lang) (if code (progn (org-html-encode-plain-text code))))))) (org-export-as (quote html) nil nil t nil))
  (ht-set! ht-temp "content" (let ((--cl-org-html-fontify-code-- (function (lambda (code lang) (if code (progn ...)))))) (org-export-as (quote html) nil nil t nil)))
  (let ((ht-temp (ht-create))) (ht-set! ht-temp "title" (or (op/read-org-option "TITLE") "Untitled")) (ht-set! ht-temp "content" (let ((--cl-org-html-fontify-code-- (function (lambda (code lang) (if code ...))))) (org-export-as (quote html) nil nil t nil))) ht-temp)
  (or param-table (let ((ht-temp (ht-create))) (ht-set! ht-temp "title" (or (op/read-org-option "TITLE") "Untitled")) (ht-set! ht-temp "content" (let ((--cl-org-html-fontify-code-- (function (lambda ... ...)))) (org-export-as (quote html) nil nil t nil))) ht-temp))
  (mustache-render (or (op/get-cache-item (if template (intern (replace-regexp-in-string "\\.mustache$" "-template" template)) :post-template)) (op/update-cache-item (if template (intern (replace-regexp-in-string "\\.mustache$" "-template" template)) :post-template) (funcall (function (lambda nil (message (concat "Read " ... " from file")) (file-to-string (concat ... ...))))))) (or param-table (let ((ht-temp (ht-create))) (ht-set! ht-temp "title" (or (op/read-org-option "TITLE") "Untitled")) (ht-set! ht-temp "content" (let ((--cl-org-html-fontify-code-- (function ...))) (org-export-as (quote html) nil nil t nil))) ht-temp)))
  op/render-content()
  (setq post-content (op/render-content))
  (progn (princ attr-plist) (setq post-content (op/render-content)) (setq assets-dir (file-name-as-directory (concat (file-name-as-directory pub-root-dir) "assets/" (replace-regexp-in-string "\\`" "" (plist-get attr-plist :uri))))) (let ((temp-buffer (generate-new-buffer " *temp*"))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (insert post-content) (beginning-of-buffer) (while (re-search-forward "<[a-zA-Z]+[^/>]+\\(src\\|href\\|data\\)=\"\\([^\"]+\\)\"[^>]*>" nil t) (setq asset-path (match-string 2)) (if (not ...) (progn ... ...)))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))) (setq component-table (let ((ht-temp (ht-create))) (ht-set! ht-temp "header" (op/render-header)) (ht-set! ht-temp "nav" (op/render-navigation-bar)) (ht-set! ht-temp "content" post-content) (ht-set! ht-temp "footer" (op/render-footer)) ht-temp)))
  (if do-pub (progn (princ attr-plist) (setq post-content (op/render-content)) (setq assets-dir (file-name-as-directory (concat (file-name-as-directory pub-root-dir) "assets/" (replace-regexp-in-string "\\`" "" (plist-get attr-plist :uri))))) (let ((temp-buffer (generate-new-buffer " *temp*"))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (insert post-content) (beginning-of-buffer) (while (re-search-forward "<[a-zA-Z]+[^/>]+\\(src\\|href\\|data\\)=\"\\([^\"]+\\)\"[^>]*>" nil t) (setq asset-path ...) (if ... ...))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))) (setq component-table (let ((ht-temp (ht-create))) (ht-set! ht-temp "header" (op/render-header)) (ht-set! ht-temp "nav" (op/render-navigation-bar)) (ht-set! ht-temp "content" post-content) (ht-set! ht-temp "footer" (op/render-footer)) ht-temp))))
  (let* ((filename (buffer-file-name)) (attr-plist (list (quote :title) (or (op/read-org-option "TITLE") "Untitled") (quote :date) (fix-timestamp-string (or (op/read-org-option "DATE") (format-time-string "%Y-%m-%d"))) (quote :mod-date) (if (not filename) (format-time-string "%Y-%m-%d") (or (op/git-last-change-date op/repository-directory filename) (format-time-string "%Y-%m-%d" (nth 5 ...)))) (quote :description) (or (op/read-org-option "DESCRIPTION") "No Description") (quote :thumb) (op/read-org-option "THUMBNAIL"))) assets-dir post-content asset-path asset-abs-path pub-abs-path converted-path component-table tags category cat-config) (setq tags (op/read-org-option "TAGS")) (if tags (progn (plist-put attr-plist :tags (delete "" (mapcar (quote trim-string) (split-string tags "[:,]+" t)))))) (if op/organization (progn (plist-put attr-plist :authororg (delete "" (mapcar (quote trim-string) (split-string (or ... user-full-name "anonymous") "[:,]+" t)))))) (setq category (funcall (or op/retrieve-category-function op/get-file-category) filename)) (plist-put attr-plist :category category) (setq cat-config (cdr (or (assoc category op/category-config-alist) (assoc "blog" op/category-config-alist)))) (plist-put attr-plist :uri (funcall (plist-get cat-config :uri-generator) (plist-get cat-config :uri-template) (plist-get attr-plist :date) (plist-get attr-plist :title))) (plist-put attr-plist :pub-dir (file-name-as-directory (concat (file-name-as-directory pub-root-dir) (replace-regexp-in-string "\\`/" "" (plist-get attr-plist :uri))))) (if do-pub (progn (princ attr-plist) (setq post-content (op/render-content)) (setq assets-dir (file-name-as-directory (concat (file-name-as-directory pub-root-dir) "assets/" (replace-regexp-in-string "\\`" "" (plist-get attr-plist :uri))))) (let ((temp-buffer (generate-new-buffer " *temp*"))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (insert post-content) (beginning-of-buffer) (while ... ... ...)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))) (setq component-table (let ((ht-temp (ht-create))) (ht-set! ht-temp "header" (op/render-header)) (ht-set! ht-temp "nav" (op/render-navigation-bar)) (ht-set! ht-temp "content" post-content) (ht-set! ht-temp "footer" (op/render-footer)) ht-temp)))) (cons attr-plist component-table))
  op/get-org-file-options("~/.op-tmp/" ("/home/trent/vegmilblog/about.org" "/home/trent/vegmilblog/index.org"))
  (setq attr-cell (op/get-org-file-options pub-root-dir (member org-file upd-list)))
  (save-current-buffer (set-buffer (setq file-buffer (or visiting (find-file org-file)))) (setq attr-cell (op/get-org-file-options pub-root-dir (member org-file upd-list))) (setq file-attr-list (cons (car attr-cell) file-attr-list)) (if (member org-file upd-list) (progn (op/publish-modified-file (cdr attr-cell) (plist-get (car attr-cell) :pub-dir)))) (if (member org-file del-list) (progn (op/handle-deleted-file org-file))))
  (lambda (org-file) (setq visiting (find-buffer-visiting org-file)) (save-current-buffer (set-buffer (setq file-buffer (or visiting (find-file org-file)))) (setq attr-cell (op/get-org-file-options pub-root-dir (member org-file upd-list))) (setq file-attr-list (cons (car attr-cell) file-attr-list)) (if (member org-file upd-list) (progn (op/publish-modified-file (cdr attr-cell) (plist-get (car attr-cell) :pub-dir)))) (if (member org-file del-list) (progn (op/handle-deleted-file org-file)))) (or visiting (kill-buffer file-buffer)))("/home/trent/vegmilblog/about.org")
  mapc((lambda (org-file) (setq visiting (find-buffer-visiting org-file)) (save-current-buffer (set-buffer (setq file-buffer (or visiting (find-file org-file)))) (setq attr-cell (op/get-org-file-options pub-root-dir (member org-file upd-list))) (setq file-attr-list (cons (car attr-cell) file-attr-list)) (if (member org-file upd-list) (progn (op/publish-modified-file (cdr attr-cell) (plist-get (car attr-cell) :pub-dir)))) (if (member org-file del-list) (progn (op/handle-deleted-file org-file)))) (or visiting (kill-buffer file-buffer))) ("/home/trent/vegmilblog/about.org" "/home/trent/vegmilblog/index.org"))
  (progn (mapc (function (lambda (org-file) (setq visiting (find-buffer-visiting org-file)) (save-current-buffer (set-buffer (setq file-buffer (or visiting ...))) (setq attr-cell (op/get-org-file-options pub-root-dir (member org-file upd-list))) (setq file-attr-list (cons (car attr-cell) file-attr-list)) (if (member org-file upd-list) (progn (op/publish-modified-file ... ...))) (if (member org-file del-list) (progn (op/handle-deleted-file org-file)))) (or visiting (kill-buffer file-buffer)))) all-list) (if (member (expand-file-name "index.org" op/repository-directory) all-list) nil (op/generate-default-index file-attr-list pub-root-dir)) (if (member (expand-file-name "about.org" op/repository-directory) all-list) nil (op/generate-default-about pub-root-dir)) (op/update-category-index file-attr-list pub-root-dir) (op/update-rss file-attr-list pub-root-dir) (op/update-tags file-attr-list pub-root-dir) (if op/organization (progn (op/update-authors file-attr-list pub-root-dir))))
  (if (or upd-list del-list) (progn (mapc (function (lambda (org-file) (setq visiting (find-buffer-visiting org-file)) (save-current-buffer (set-buffer (setq file-buffer ...)) (setq attr-cell (op/get-org-file-options pub-root-dir ...)) (setq file-attr-list (cons ... file-attr-list)) (if (member org-file upd-list) (progn ...)) (if (member org-file del-list) (progn ...))) (or visiting (kill-buffer file-buffer)))) all-list) (if (member (expand-file-name "index.org" op/repository-directory) all-list) nil (op/generate-default-index file-attr-list pub-root-dir)) (if (member (expand-file-name "about.org" op/repository-directory) all-list) nil (op/generate-default-about pub-root-dir)) (op/update-category-index file-attr-list pub-root-dir) (op/update-rss file-attr-list pub-root-dir) (op/update-tags file-attr-list pub-root-dir) (if op/organization (progn (op/update-authors file-attr-list pub-root-dir)))))
  (let* ((upd-list (plist-get change-plist :update)) (del-list (plist-get change-plist :delete)) visiting file-buffer attr-cell file-attr-list) (if (or upd-list del-list) (progn (mapc (function (lambda (org-file) (setq visiting (find-buffer-visiting org-file)) (save-current-buffer (set-buffer ...) (setq attr-cell ...) (setq file-attr-list ...) (if ... ...) (if ... ...)) (or visiting (kill-buffer file-buffer)))) all-list) (if (member (expand-file-name "index.org" op/repository-directory) all-list) nil (op/generate-default-index file-attr-list pub-root-dir)) (if (member (expand-file-name "about.org" op/repository-directory) all-list) nil (op/generate-default-about pub-root-dir)) (op/update-category-index file-attr-list pub-root-dir) (op/update-rss file-attr-list pub-root-dir) (op/update-tags file-attr-list pub-root-dir) (if op/organization (progn (op/update-authors file-attr-list pub-root-dir))))))
  op/publish-changes(("/home/trent/vegmilblog/about.org" "/home/trent/vegmilblog/index.org") (:update ("/home/trent/vegmilblog/about.org" "/home/trent/vegmilblog/index.org") :delete nil) "~/.op-tmp/")
  (let* ((orig-branch (op/git-branch-name op/repository-directory)) (to-repo (not (stringp pub-base-dir))) (store-dir (if to-repo "~/.op-tmp/" pub-base-dir)) changed-files all-files remote-repos) (op/git-change-branch op/repository-directory op/repository-org-branch) (op/prepare-theme store-dir) (setq all-files (op/git-all-files op/repository-directory)) (setq changed-files (if force-all (cons (quote :update) (cons all-files (quote (:delete nil)))) (op/git-files-changed op/repository-directory (or base-git-commit "HEAD~1")))) (op/publish-changes all-files changed-files store-dir) (if to-repo (progn (op/git-change-branch op/repository-directory op/repository-html-branch) (copy-directory store-dir op/repository-directory t t t) (delete-directory store-dir t))) (if (and to-repo auto-commit) (progn (op/git-commit-changes op/repository-directory "Update published html files, committed by org-page.") (if auto-push (progn (setq remote-repos (op/git-remote-name op/repository-directory)) (if (not remote-repos) (message "No valid remote repository found.") (let (repo) (if ... ... ...) (if ... ... ...))))) (op/git-change-branch op/repository-directory orig-branch))) (if to-repo (message "Publication finished: on branch '%s' of repository '%s'." op/repository-html-branch op/repository-directory) (message "Publication finished, output directory: %s." pub-base-dir)))
  op/do-publication(t nil nil t)
  eval((op/do-publication t nil nil t) nil)
  eval-last-sexp-1(t)
  eval-last-sexp(t)
  eval-print-last-sexp()
  call-interactively(eval-print-last-sexp nil nil)
